mlflow/R/mlflow/R/model-h2o.R

```diff
@@ -0,0 +1,74 @@
+#' @include model-utils.R
+NULL
+
+#' @rdname mlflow_save_model
+#' @export
+mlflow_save_model.H2OModel <- function(model,
+                                       path,
+                                       model_spec = list(),
+                                       conda_env = NULL,
+                                       ...) {
+  assert_pkg_installed("h2o")
+
+  if (dir.exists(path)) unlink(path, recursive = TRUE)
+  dir.create(path, recursive = TRUE)
+  path <- normalizePath(path)
+
+  model_data_subpath <- "model.h2o"
+  model_data_path <- file.path(path, model_data_subpath)
+
+  dir.create(model_data_path)
+
+  h2o_save_location <- h2o::h2o.saveModel(
+    object = model, path = model_data_path, force = TRUE
+  )
+  model_file <- basename(h2o_save_location)
+
+  settings <- list(
+    full_file = h2o_save_location,
+    model_file = model_file,
+    model_dir = model_data_path
+  )
+  yaml::write_yaml(settings, file.path(model_data_path, "h2o.yaml"))
+
+  # needed because R and python packages may not have the same patch number
+  version <- remove_patch_version(
+    as.character(utils::packageVersion("h2o"))
+  )
+
```

Comment:
we did not require exact match on patch versions (or at least that was the plan originally : ) )